src/toil/fileStore.py

```diff
@@ -721,6 +721,13 @@ def readGlobalFile(self, fileStoreID, userPath=None, cache=True, mutable=None):
                                                               0.0, False)
         return localFilePath
 
+    def exportFile(self, jobStoreFileID, dstUrl):
+        while jobStoreFileID in self._pendingFileWrites:
```

Comment:
I see. So far it seems like this approach would still be cleaner since it is further abstracted from the actual caching logic. I don't have to refactor/rename any methods and I don't have to mess with the harbinger/lock file at all, I can just perform my check with the fileID directly.